discretisedfield/util/util.py

```diff
@@ -40,7 +40,11 @@ def _(val, mesh, dim, dtype):
                          f' expected dimension is {dim}')
     if dtype is None:
         dtype = max(np.asarray(val).dtype, np.float64)
-    return np.full((*mesh.n, dim), val, dtype=dtype)
+
+    if isinstance(val, np.ndarray) and val.shape == mesh.n + (dim,):
+        return np.asarray(val, dtype=dtype)  # no int arrays
+    else:
+        return np.full((*mesh.n, dim), val, dtype=dtype)
```

Comment:
@marijanbeg Again thinking about this I was wondering whether it might be saver to always create a copy of the data (what `np.full` does), i.e. not changing anything here. Any thoughts?